src/ghidra/p_code_extractor/PcodeExtractor.java

```diff
@@ -1,288 +1,249 @@
-            handleEmptyBlock(codeBlock);
+            callingConventions.put(prototypeModel.getName(), cconv);
         }
 
-        return PcodeBlockData.blocks;
-    }
+        // assembling everything together
+        ProjectSimple project = new ProjectSimple(functions, registerProperties, cpuArch, stackPointerRegister,
+                callingConventions, dataTypeProperties, imageBase);
 
+        // serialization
 
-    /**
-     * 
-     * @param codeBlock: Current empty block
-     * @return New jmp term containing fall through address
-     * 
-     * Adds fallthrough address jump to empty block if available
-     */
-    protected void handleEmptyBlock(CodeBlock codeBlock) {
+        Gson gson = new GsonBuilder().setPrettyPrinting().serializeNulls().create();
         try {
-            CodeBlockReferenceIterator destinations = codeBlock.getDestinations(getMonitor());
-            if(destinations.hasNext()) {
-                Tid jmpTid = new Tid(String.format("instr_%s_%s", codeBlock.getFirstStartAddress().toString(), 0), codeBlock.getFirstStartAddress().toString());
-                Tid gotoTid = new Tid();
-                String destAddr = destinations.next().getDestinationBlock().getFirstStartAddress().toString();
-                gotoTid.setId(String.format("blk_%s", destAddr));
-                gotoTid.setAddress(destAddr);
-                PcodeBlockData.blocks.get(0).getTerm().addJmp(new Term<Jmp>(jmpTid, new Jmp(ExecutionType.JmpType.GOTO, "BRANCH", new Label((Tid) gotoTid), 0)));
-            }
-        } catch (CancelledException e) {
-            System.out.printf("Could not retrieve destinations for block at: %s\n", codeBlock.getFirstStartAddress().toString());
+            FileWriter writer = new FileWriter(getScriptArgs()[0]);
+            gson.toJson(project, writer);
+            writer.close();
+        } catch (JsonIOException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
         }
-    }
 
+        println("Pcode was successfully extracted!");
+    }
 
     /**
+     * Returns list of VarnodeSimple of integer parameter in order defined by the
+     * calling convention.
      * 
-     * Checks whether the assembly instruction is a nop instruction and adds a jump to the block.
-     * Checks whether a jump occured within a ghidra generated pcode block and fixes the control flow
-     * by adding missing jumps between artificially generated blocks.
-     * Checks whether an instruction is in a delay slot and, if so, ignores it 
-     * as Ghidra already includes the instruction before the jump
+     * This functions returns the base register, e.g. RDI is returned and not EDI.
+     * *Approximation*: This function returns the first 10 or less integer registers
+     * This is due to the only way of extracting the integer parameter register
+     * available
+     * (https://ghidra.re/ghidra_docs/api/ghidra/program/model/lang/PrototypeModel.html)
```

Comment:
We definitely have to revisit this when the Rust-side of the parsing is implemented. Just to make sure whether we want to fall back to the old "directly parse the xml file" behaviour or not.

Reply:
I will keep that in mind. Nevertheless it should come up during parsing the pcode.